packages/simulation/tests/convertNFAtoDFA.test.ts

```diff
@@ -156,4 +157,51 @@ describe('Check to ensure DFA graph is displayed as expected', () => {
+    expect(graph.transitions.filter((transition) => transition.from === 0).length).toBe(7)
+    expect(graph.transitions.filter((transition) => transition.from === 0 && transition.read === 'C')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 0 && transition.read === 'D')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 0 && transition.read === 'E')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 0 && transition.read === 'F')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 0 && transition.read === 'G')[0].to).toBe(2)
+
+    expect(graph.transitions.filter((transition) => transition.from === 1).length).toBe(7)
+    expect(graph.transitions.filter((transition) => transition.from === 1 && transition.read === 'A')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 1 && transition.read === 'B')[0].to).toBe(2)
+
+    // This is the trap state, should go to itself for every symbol
+    expect(graph.transitions.filter((transition) => transition.from === 2).length).toBe(7)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'A')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'B')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'C')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'D')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'E')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'F')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 2 && transition.read === 'G')[0].to).toBe(2)
+
+    // Final state with no transitions to must all go to a single trap state
+    expect(graph.transitions.filter((transition) => transition.from === 3).length).toBe(7)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'A')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'B')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'C')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'D')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'E')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'F')[0].to).toBe(2)
+    expect(graph.transitions.filter((transition) => transition.from === 3 && transition.read === 'G')[0].to).toBe(2)
```

Comment:
```suggestion
    const isTrapped = (from: number, toTrap: string) => {
      const transitions = graph.transitions.filter(t => t.from === from)
      expect(transitions.length).toBe(alphabet.length)
      // Find all the transitions that read a symbol we expect to go to a trap state
      const expectedTrapTransitions = transitions
        .filter(t => toTrap.includes(t.read))
        .map(t => t.to)
      expect(new Set(expectedTrapTransitions).size).toBe(1)
      // All trap transitions should be going to state '2' (The trap state)
      expect(expectedTrapTransitions[0]).toBe(2)
    }
    // Transitions that are not initially present should all go to a single trap state "2"
    isTrapped(0, 'CDEFG')
    isTrapped(1, 'AB')
    // This is the trap state, should go to itself for every symbol
    isTrapped(2, alphabet)
    // Final state with no transitions to must all go to a single trap state
    isTrapped(3, alphabet)
```

Try and group similar logic into functions. Espically for tests because if behavior changes in future then it is good if less spots need to be changed